### PR TITLE
fix: Sending multiple UAC events for same value UA

### PR DESCRIPTION
**Pull request by mmustafa-tse**
## Summary
A client reported that UAC events are forwarded on initialization everytime for some attributes even though the values are not changed. Per investigating it seemed that values that are set to false are only affected by this issue, furthermore I was able to also confirm that values such as 0 and an empty string '' would have the same issue since Javascript recognize them as null values. The code change here is to only change previousUserAttributeValue when it comes in as undefined since this is the value when it is not seen before.

## Testing Plan
Tested locally with overrides in sample app and also added unit testing case

### CODE REVIEW

**Comment by mmustafa-tse** [comment]
@rmi22186

**Comment by rmi22186** [comment]
Thanks @mmustafa-tse!  You should actually make the change inside of the `src/identity.js` folder.  Please undo the change to dist.  Once a PR is merged, our release process automatically builds the src files into the dist folder.

**Comment by mmustafa-tse** [comment]
@rmi22186 - changes requested done, thanks for the clarification!

File: src/identity.js [hunk 1]
```diff
@@ -1819,7 +1819,7 @@ export default function Identity(mpInstance) {
         deleted,
         user
     ) {
-        if (!previousUserAttributeValue) {
+        if (previousUserAttributeValue === undefined) {
```
Comment (alexs-mparticle) [inline src/identity.js:1822]: I think this is a better way to check to make sure that this variable is actually defined.

```suggestion
        if (typeof previousUserAttributeValue === 'undefined') {
```

**Comment by mmustafa-tse** [comment]
@alexs-mparticle Agree, changes committed. Thanks!

File: test/src/tests-identities-attributes.js [hunk 2]
```diff
@@ -1163,4 +1163,46 @@ describe('identities and attributes', function() {
 
         done();
     });
+
+    it('should not send user attribute change requests when user attribute already set with same value', function(done) {
```
Comment (rmi22186) [inline test/src/tests-identities-attributes.js:1167]: This test would pass even with the original code for `if (!previousUserAttributeValue)`.  Please add tests for where the attribute is `0`, `false`, and `undefined`.

**Review by rmi22186 — CHANGES_REQUESTED** [review]

File: test/src/tests-identities-attributes.js (hunk 2 above)
Comment (mmustafa-tse) [reply test/src/tests-identities-attributes.js:1167]: @rmi22186 should I also remove the test case for attribute age: 25?

Comment (rmi22186) [reply test/src/tests-identities-attributes.js:1167]: i think you can add a few user attributes, so it's fine to keep them and just add the others.

**Comment by mmustafa-tse** [comment]
@rmi22186 testing cases for 0, '' and  false added.

**Comment by rmi22186** [comment]
This looks good, but I would add one more test.  In the case where you are going from one false-y value to another, let's confirm that a UAC is sent since going from `0` to `''` to `false` are actually different values.  Let's make this a new test also, and just do something like:

`should send user attributes when setting different falsey values`:
1. set a user attribute to 0
2. set the same  user attribute to `''`
3. check that the event exists and is a UAC
4. set the same user attribute to `false`
5. check that the event exists and is a UAC
6. set the same  user attribute back to `0`
7. 5. check that the event exists and is a UAC

File: test/src/tests-identities-attributes.js [hunk 3]
```diff
@@ -1163,4 +1163,171 @@ describe('identities and attributes', function() {
...
+        mParticle.Identity.getCurrentUser().setUserAttribute('testFalse', false);
+        mParticle.Identity.getCurrentUser().setUserAttribute('testEmptyString', '');
+        mParticle.Identity.getCurrentUser().setUserAttribute('testZero', 0);
+        (window.fetchMock.lastOptions() === undefined).should.equal(true);
+        (window.fetchMock._calls.length === 0).should.equal(true);
+
+        done()
+    });
+
+    it('should send user attributes when setting different falsey values', function(done) {
```
Comment (rmi22186) [inline test/src/tests-identities-attributes.js:1259]:
```suggestion
    it('should send user attribute change event when setting different falsey values', function(done) {
```

File: src/identity.js [hunk 4]
```diff
@@ -1830,7 +1830,7 @@ export default function Identity(mpInstance) {
                     userAttributeChanges: {
                         UserAttributeName: key,
                         New: newValue,
-                        Old: previousUserAttributeValue || null,
+                        Old: previousUserAttributeValue,
```
Comment (rmi22186) [inline src/identity.js:1833]: Any reason for removing the null here?

Comment (mmustafa-tse) [reply src/identity.js:1833]: Another little bug noticed when testing, this is causing the 'old' values in the event batches to be null instead showing their value for the case of falsey attributes. Let me know if this needs to be changed back but isn't the only time we want old to be null when it's never seen before? Hence the condition on line 1822

File: test/src/tests-identities-attributes.js (hunk 3 above)
Comment (mmustafa-tse) [reply test/src/tests-identities-attributes.js]: @rmi22186 done

File: src/identity.js (hunk 4 above)
Comment (rmi22186) [reply src/identity.js:1833]: great catch

**Review by rmi22186 — APPROVED** [review]
